examples/3d/reflection_probes.rs

```diff
@@ -0,0 +1,270 @@
+    // reflection probe is shown in the sphere.
+    #[default]
+    ReflectionProbe = 2,
+}
+
+// The various reflection maps.
+#[derive(Resource)]
+struct Cubemaps {
+    // The blurry diffuse cubemap. This is used for both the world environment
+    // map and the reflection probe. (In reality you wouldn't do this, but this
+    // reduces complexity of this example a bit.)
+    diffuse: Handle<Image>,
+
+    // The specular cubemap that reflects the world, but not the cubes.
+    specular_environment_map: Handle<Image>,
+
+    // The specular cubemap that reflects both the world and the cubes.
+    specular_reflection_probe: Handle<Image>,
+
+    // The skybox cubemap image. This is almost the same as
+    // `specular_environment_map`.
+    skybox: Handle<Image>,
+}
+
+fn main() {
+    // Create the app.
+    App::new()
+        .add_plugins(DefaultPlugins)
+        .init_resource::<ReflectionMode>()
+        .init_resource::<Cubemaps>()
+        .add_systems(Startup, setup)
+        .add_systems(PreUpdate, add_environment_map_to_camera)
+        .add_systems(Update, change_reflection_type)
+        .add_systems(Update, rotate_camera)
+        .run();
+}
+
+// Initializes the scene.
+fn setup(
+    mut commands: Commands,
+    mut meshes: ResMut<Assets<Mesh>>,
+    mut materials: ResMut<Assets<StandardMaterial>>,
+    asset_server: Res<AssetServer>,
+    reflection_mode: Res<ReflectionMode>,
+    cubemaps: Res<Cubemaps>,
+) {
+    // Spawn the cubes, light, and camera.
+    commands.spawn(SceneBundle {
+        scene: asset_server.load("models/cubes/Cubes.glb#Scene0"),
+        ..SceneBundle::default()
+    });
+
+    // Create a sphere mesh.
+    let sphere_mesh = meshes.add(
+        Mesh::try_from(shape::Icosphere {
+            radius: 1.0,
+            subdivisions: 7,
+        })
+        .unwrap(),
+    );
+
+    // Create a sphere.
+    commands.spawn(PbrBundle {
+        mesh: sphere_mesh.clone(),
+        material: materials.add(StandardMaterial {
+            base_color: Color::hex("#ffd891").unwrap(),
+            metallic: 1.0,
+            perceptual_roughness: 0.0,
+            ..StandardMaterial::default()
+        }),
+        transform: Transform::default(),
+        ..PbrBundle::default()
+    });
+
+    // Create the reflection probe.
+    create_reflection_probe(&mut commands, &cubemaps);
+
+    // Create the text.
+    commands.spawn(
+        TextBundle {
+            text: create_text(*reflection_mode, &asset_server),
+            ..TextBundle::default()
+        }
+        .with_style(Style {
+            position_type: PositionType::Absolute,
+            bottom: Val::Px(10.0),
+            left: Val::Px(10.0),
+            ..default()
+        }),
+    );
+}
+
+// Creates the reflection probe, which is a box that refers to an environment
+// map and encloses the objects that are to be affected by it (in this case, the
+// sphere).
+fn create_reflection_probe(commands: &mut Commands, cubemaps: &Cubemaps) {
```

Comment:
Please add a ReflectionProbeBundle, consisting of spatial bundle, light probe, and environment map light.